packages/project/src/db/props.ts

```diff
@@ -1,22 +1,73 @@
 import {
   type Tree,
   type AllUserProps,
-  UserProps,
+  type UserProp,
+  // TODO: move zod schema into project package
+  UserUserProps,
+  UserDbProps,
 } from "@webstudio-is/react-sdk";
 import { applyPatches, type Patch } from "immer";
 import { prisma } from "@webstudio-is/prisma-client";
+import { formatAsset } from "@webstudio-is/asset-uploader/server";
 
 export const loadByTreeId = async (treeId: Tree["id"]) => {
   const instancePropsEntries = await prisma.instanceProps.findMany({
     where: { treeId },
   });
 
+  // Fin all assetId in all props
+  const assetIds: string[] = [];
+
+  for (const instanceProps of instancePropsEntries) {
+    const props = JSON.parse(instanceProps.props);
+    const dpProps = UserDbProps.parse(props);
```

Comment:
just call it parsedProps
```suggestion
    const parsedProps = UserDbProps.parse(props);
```

Reply:
cleaned it up for you

```ts
return instancePropsEntries.map((instanceProps) => {
  const props = JSON.parse(instanceProps.props);
  // Can be changed onto `props as ` having that above is already checked, no optmizations now
  const parsedProps = UserDbProps.parse(props);
  const resultProps: UserProp[] = [];

  for (const parsedProp of parsedProps) {
    if (parsedProp.assetId == null) {
      resultProps.push(parsedProp);
      continue;
    }
    
    const asset = assetsMap.get(dbProp.assetId);
    if (asset) {
      resProps.push({
        ...dbProp,
        asset,
      });
      continue
    } 
    // Can be ok if asset was deleted having that no db constraint checks exists
    // @todo: remove throw on something else
    throw new Error(`Asset with assetId "${dbProp.assetId}" was not found`);
  }

  return {
    ...instanceProps,
    props: resultProps,
  };
});
```

Reply:
IMO dbProps is perfectly fine as gives information that this props are somehow different from userProps.
Changed resultProps onto userProps to make it more visible